src/coreclr/jit/codegenarm64.cpp

```diff
@@ -3846,17 +3846,28 @@ void CodeGen::genEmitHelperCall(unsigned helper, int argSize, emitAttr retSize,
 void CodeGen::genSIMDIntrinsic(GenTreeSIMD* simdNode)
 {
     // NYI for unsupported base types
-    if (simdNode->GetSimdBaseType() != TYP_INT && simdNode->GetSimdBaseType() != TYP_LONG &&
-        simdNode->GetSimdBaseType() != TYP_FLOAT && simdNode->GetSimdBaseType() != TYP_DOUBLE &&
-        simdNode->GetSimdBaseType() != TYP_USHORT && simdNode->GetSimdBaseType() != TYP_UBYTE &&
-        simdNode->GetSimdBaseType() != TYP_SHORT && simdNode->GetSimdBaseType() != TYP_BYTE &&
-        simdNode->GetSimdBaseType() != TYP_UINT && simdNode->GetSimdBaseType() != TYP_ULONG)
-    {
-        // We don't need a base type for the Upper Save & Restore intrinsics, and we may find
-        // these implemented over lclVars created by CSE without full handle information (and
-        // therefore potentially without a base type).
-        if ((simdNode->gtSIMDIntrinsicID != SIMDIntrinsicUpperSave) &&
+    if (!TypeIsInList(simdNode->GetSimdBaseType(), TYP_INT, TYP_LONG, TYP_FLOAT, TYP_DOUBLE, TYP_USHORT, TYP_UBYTE,
+                      TYP_SHORT, TYP_BYTE, TYP_UINT, TYP_ULONG))
+    {
+        bool unsupportedBaseType = true;
+
+        if ((simdNode->gtSIMDIntrinsicID == SIMDIntrinsicUpperSave) ||
             (simdNode->gtSIMDIntrinsicID != SIMDIntrinsicUpperRestore))
+        {
+            // We don't need a base type for the Upper Save & Restore intrinsics, and we may find
+            // these implemented over lclVars created by CSE without full handle information (and
+            // therefore potentially without a base type).
+            unsupportedBaseType = false;
+        }
+
+        const GenTree* op1 = simdNode->gtGetOp1();
+        if ((simdNode->gtSIMDIntrinsicID == SIMDIntrinsicInit) && (op1->IsIntegralConst(0) || op1->IsFPZero()))
```

Comment:
We have `GenTree::IsSIMDZero` that should be preferred here, since it will be updated to include the `get_Zero` HWIntrinsics in the future: <https://github.com/dotnet/runtime/blob/main/src/coreclr/jit/gentree.h#L7061>